### PR TITLE
feat: modify for custom CA certificates

### LINKED ISSUE #2325: Enable custom root CA renku-graph

### PR DESCRIPTION
**Pull request by olevski**
Allows the KG services to properly access another service (i.e. namely gitlab and/or keycloak) if that service is using self-signed certificates.

I followed the Gitlab cloud native helm chart and how they tackle this issue when addressing this. See for examples:

https://gitlab.com/gitlab-org/charts/gitlab/-/blob/master/templates/_certificates.tpl
https://gitlab.com/gitlab-org/charts/gitlab/-/blob/master/charts/gitlab/charts/webservice/templates/deployment.yaml#L102

Fixes SwissDataScienceCenter/renku#2325

Furthermore, this makes the changes made in https://github.com/SwissDataScienceCenter/renku-graph/pull/509 not needed. This is how injecting the certificates is handled across all other services so it is better to do it the same way here. Also I think it is nicer if handling certificates is more isolated from the business logic / application code.

/deploy #persist

### CODE REVIEW

**Comment by olevski** [comment]
The dependent PRs have been resolved. This is ready for review.

**Comment by jachro** [comment]
Hi Tasko,

I've just pushed a huge change to our `master`. Could you try to pull
master on top of your branch, please?

FYI, I believe it might be better to use our `development` branch as a
starting point for any new work.

**Comment by olevski** [comment]
Ok I brought this up to speed with master @jachro.

File: helm-chart/renku-graph/templates/event-log-deployment.yaml
Comment (jachro) [inline helm-chart/renku-graph/templates/event-log-deployment.yaml:91]: EL is running on openjdk15 too now

File: helm-chart/renku-graph/templates/knowledge-graph-deployment.yaml
Comment (jachro) [inline helm-chart/renku-graph/templates/knowledge-graph-deployment.yaml:84]: jdk15, too if that makes a difference

File: helm-chart/renku-graph/templates/token-repository-deployment.yaml
Comment (jachro) [inline helm-chart/renku-graph/templates/token-repository-deployment.yaml:93]: Same as for the others. Actually, all KG services are running OpenJDK 15 now :)

File: helm-chart/renku-graph/templates/webhook-service-deployment.yaml
Comment (jachro) [inline helm-chart/renku-graph/templates/webhook-service-deployment.yaml:84]: ditto

**Comment by jachro** [comment]
Just a question as things are coming back to me from the time when I was doing the initial implementation for the custom CA support. I remember I had to install the `pem` file so `git` could connect to GitLab. I guess you've probably taken care of that but just to be sure.

**Comment by olevski** [comment]
> Just a question as things are coming back to me from the time when I was doing the initial implementation for the custom CA support. I remember I had to install the `pem` file so `git` could connect to GitLab. I guess you've probably taken care of that but just to be sure.

@jachro yeah I inject the custom certificate in the linux system /etc/ssl/certs (in addition to the java cert store). And git looks at this and respects this.

**Review by jachro — DISMISSED** [review]
Thanks Tasko for your contribution to the Graph :) You are more than welcome to do more ;)

**Comment by olevski** [comment]
Please do not merge this. It seems that there is most like a problem with the acceptance tests but I want to confirm before ignoring the acceptance tests.

**Comment by olevski** [comment]
@jachro this is ready now. The acceptance tests even pass 😄 

I had to dismiss your approval to get the branch up to date and resolve conflicts.

**Review by jachro — APPROVED** [review]

**Pull request merged**
